WordPress/Classes/Utility/CoreDataManager.swift

```diff
@@ -0,0 +1,401 @@
+    private init() {
+        observe()
+    }
+
+    private lazy var writerContext: NSManagedObjectContext = {
+        let context = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+        context.persistentStoreCoordinator = self.persistentStoreCoordinator
+        return context
+    }()
+
+    lazy var mainContext: NSManagedObjectContext = {
+        let context = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
+        context.parent = self.writerContext
+        return context
+    }()
+
+    var persistentStoreCoordinator: NSPersistentStoreCoordinator {
+        persistentContainer.persistentStoreCoordinator
+    }
+
+    var managedObjectModel: NSManagedObjectModel {
+        guard let mom = NSManagedObjectModel(contentsOf: modelURL) else {
+            fatalError("Could not load model")
+        }
+        return mom
+    }
+
+    // Error handling
+    private lazy var sentryStartupError: SentryStartupEvent = {
+        return SentryStartupEvent()
+    }()
+
+    lazy var persistentContainer: NSPersistentContainer = {
+        migrateDataModelIfNecessary()
+
+        let container = NSPersistentContainer(name: Constants.name, managedObjectModel: managedObjectModel)
+        container.persistentStoreDescriptions = [storeDescription]
+
+        container.loadPersistentStores { [weak self] (storeDescription, error) in
+            guard let `self` = self, let error = error else {
+                return
+            }
+            DDLogError("⛔️ [CoreDataManager] loadPersistentStore failed. Attempting to recover... \(error)")
+            self.sentryStartupError.add(error: error)
+
+            /// Backup the old Store
+            ///
+            do {
+                try CoreDataIterativeMigrator.backupDatabase(at: self.storeURL)
+            } catch {
+                self.sentryStartupError.add(error: error)
+                self.sentryStartupError.send(title: "Can't initialize Core Data stack")
+                fatalError("☠️ [CoreDataManager] Cannot backup Store: \(error)")
+            }
+
+            /// Retry!
+            ///
+            container.loadPersistentStores { [weak self] (storeDescription, error) in
+                guard let error = error as NSError? else {
+                    return
+                }
+
+                self?.sentryStartupError.add(error: error)
+                self?.sentryStartupError.send(title: "Can't initialize Core Data stack")
+                fatalError("☠️ [CoreDataManager] Recovery Failed! \(error) [\(error.userInfo)]")
+            }
+        }
+        return container
+    }()
+
+    private func childContext(with concurrencyType: NSManagedObjectContextConcurrencyType) -> NSManagedObjectContext {
+        let childManagedObjectContext = NSManagedObjectContext(concurrencyType: concurrencyType)
+        childManagedObjectContext.parent = self.mainContext
+        childManagedObjectContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
```

Comment:
Maybe.. It's been implemented like this, for now, as it's the Swift-migrated version of `newChildContextWithConcurrencyType` in `ContextManmager.m`, which contains the same policy. Should we change that one too?

Reply:
I'm honestly not sure. @astralbodies ? Thoughts?

Reply:
Honestly if I were to write this from scratch I'd probably pick `NSMergeByPropertyObjectTrumpMergePolicy` because what's in memory is probably "newer" than what's on disk.

Reply:
I've gone back through the older code. Looks like we added the NSMergeByPropertyObjectTrumpMergePolicy in https://github.com/wordpress-mobile/WordPress-iOS/pull/1183 to deal with a bunch of merge errors we were seeing logged. 

Removing the policy might reveal a new set of merge issues that we're unaware of because the policy has been handling them.  This might be nice to do for developer builds (fix all the things!) but I think I've come around to keeping the policy for production builds since (as @Gio2018 pointed out) its been like that for a while now. :)   Maybe we can revisit in the future.